src/primitive.ts

```diff
@@ -126,51 +126,66 @@ export function parseString (str: string, ptr = 0, endPtr = str.length): string
 	return parsed + str.slice(sliceStart, endPtr - 1)
 }
 
-export function parseValue (value: string, toml: string, ptr: number): boolean | number | TomlDate {
+export enum IntegerParsing { NUMBER_OR_ERROR, NUMBER_OR_BIGINT, BIGINT_ONLY }
+
+export function parseValue(value: string, toml: string, ptr: number, integerParsing: IntegerParsing = IntegerParsing.NUMBER_OR_ERROR): boolean | number | bigint | TomlDate {
 	// Constant values
-	if (value === 'true') return true
-	if (value === 'false') return false
-	if (value === '-inf') return -Infinity
-	if (value === 'inf' || value === '+inf') return Infinity
-	if (value === 'nan' || value === '+nan' || value === '-nan') return NaN
+	if (value === 'true') return true;
+	if (value === 'false') return false;
+	if (value === '-inf') return -Infinity;
+	if (value === 'inf' || value === '+inf') return Infinity;
+	if (value === 'nan' || value === '+nan' || value === '-nan') return NaN;
 
-	if (value === '-0') return 0 // Avoid FP representation of -0
+	if (value === '-0') return 0; // Avoid FP representation of -0
 
 	// Numbers
-	let isInt
-	if ((isInt = INT_REGEX.test(value)) || FLOAT_REGEX.test(value)) {
+	const isInt = INT_REGEX.test(value);
+	const isFloat = FLOAT_REGEX.test(value);
```

Comment:
I use `let` because it's shorter than `const`, I think most if not all variables are `let` and not `const`.

Also, storing both is wasteful: given that we enter the branch only of it is an integer or a float, if `isInt` is false then you know it must be a floating point value. This is why it was written this way before, and I don't think modifying it is necessary at all.

Reply:
Done

Reply:
My point is that if you know `isInt`, then you necessarily know `isFloat` when you're inside the numeric processing logic. `isFloat === !isInt`, so there's no need to store it alongside `isInt`